### PR TITLE
Setup Windows x64/arm64 GitHub action for Classic Flang LLVM

### PR DESCRIPTION
**Pull request by kaadam**
* Added a GitHub hosted Windows action to produce proper binaries for Flang.
* Enable running tests on Windows x64
* Updated install Python step, since CMake chooses the most recent version of 32bit Python libraries, which is 3.11 on 'windows-latest' GitHub image. Using 32bit version of Python causes a 'memory error' during the tests.

```
[856/859] Building RC object unittests\tools\llvm-mca\CMake-- Testing: 58046 tests, 2 workers --
Exception in thread Thread-13 (_handle_results):
Traceback (most recent call last):
  File "C:\hostedtoolcache\windows\Python\3.11.1\x86\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
  File "C:\hostedtoolcache\windows\Python\3.11.1\x86\Lib\threading.py", line 975, in run
    self._target(*self._args, **self._kwargs)
  File "C:\hostedtoolcache\windows\Python\3.11.1\x86\Lib\multiprocessing\pool.py", line 579, in _handle_results
    task = get()
           ^^^^^
  File "C:\hostedtoolcache\windows\Python\3.11.1\x86\Lib\multiprocessing\connection.py", line 249, in recv
    buf = self._recv_bytes()
          ^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.11.1\x86\Lib\multiprocessing\connection.py", line 317, in _recv_bytes
    return self._get_more_data(ov, maxsize)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\hostedtoolcache\windows\Python\3.11.1\x86\Lib\multiprocessing\connection.py", line 339, in _get_more_data
    ov, err = _winapi.ReadFile(self._handle, left, overlapped=True)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
MemoryError
```

### CODE REVIEW

File: .github/workflows/pre-compile_llvm_windows.yml
Comment (bryanpkc) [inline .github/workflows/pre-compile_llvm_windows.yml]: `.github/workflows/llvm-project-tests.yml` uses the following; consider sticking to it:
```yaml
uses: llvm/actions/install-ninja@main
```

**Comment by bryanpkc** [comment]
Why doesn't this job run any test?

**Comment by kaadam** [comment]
> Why doesn't this job run any test?

Yes, that's a good point. I haven't included yet. Running llvm+clang tests would be enough? Shall I use existing action for that?

**Comment by bryanpkc** [comment]
> Yes, that's a good point. I haven't included yet. Running llvm+clang tests would be enough? Shall I use existing action for that?

My suggestion is to add your current code into `pre-compile_llvm.yml`, which is also designed to run after a successful merge and create binaries that have passed PR reviews and CI tests. For testing PRs, I suggest you add Windows to the matrix in `llvm-project-tests.yml`.

File: .github/workflows/llvm-project-tests.yml
Comment (bryanpkc) [inline .github/workflows/llvm-project-tests.yml]: The newly added code and comment now conflict with the original comment above. Please merge and rewrite both comments.

**Comment by bryanpkc** [comment]
Have you tested the `build-win64` job in `pre-compile_llvm.yml` in your fork? If you have, can you paste the link to your job output? Since this job runs post-merge, it is not possible to test its configuration when reviewing this pull request.

**Comment by kaadam** [comment]
> Have you tested the `build-win64` job in `pre-compile_llvm.yml` in your fork? If you have, can you paste the link to your job output? Since this job runs post-merge, it is not possible to test its configuration when reviewing this pull request.

Bryan, yes, sorry I forgot to place the link here:
This was the latest run: https://github.com/kaadam/classic-flang-llvm-project/actions/runs/4088129300

File: .github/workflows/llvm-project-tests.yml
Comment (kaadam) [reply .github/workflows/llvm-project-tests.yml]: Okay, I'm updating the comment.

**Comment by kaadam** [comment]
@bryanpkc I'm thinking to add a new Windows x64 action for creating prebuilt LLVM to Windows on ARM. However we cannot run tests on it. What do you think? Is that acceptable if I add an only cross-build job?

**Comment by bryanpkc** [comment]
@kaadam Is the Windows on ARM build always cross-compiled? Can we build Flang natively on Windows on ARM? If so, it would be possible to create a self-hosted runner for the latter.

If that's difficult, I don't have a problem with the cross-compilation job. Maybe in a separate PR though?

**Review by bryanpkc — APPROVED** [review]

**Comment by kaadam** [comment]
> @kaadam Is the Windows on ARM build always cross-compiled? Can we build Flang natively on Windows on ARM? If so, it would be possible to create a self-hosted runner for the latter.
> 
> If that's difficult, I don't have a problem with the cross-compilation job. Maybe in a separate PR though?

@bryanpkc  Basically Flang build is running on Windows on ARM natively, we only cross-compile classic-llvm for Windows on ARM, since the build was faster on a x64 host machine. Now we have a proper hardware to do both build natively. So I will create a self-hosted runner for WoA build.

**Comment by bryanpkc** [comment]
@kaadam Is this PR ready to merge then? And you will submit another PR for the Windows-on-ARM builds?

**Comment by kaadam** [comment]
> @kaadam Is this PR ready to merge then? And you will submit another PR for the Windows-on-ARM builds?

@bryanpkc I did a tiny modification on this PR, It's ready for review/merge now. Yes, I will upload the Windows on ARM configuration in a follow-up PR.

**Comment by kaadam** [comment]
@bryanpkc I could simplify the x64 and arm64 workflow, so maybe better if I squash two PRs into one.

**Comment by kaadam** [comment]
@bryanpkc I've updated the PR. Could you look at it again please? Currently Windows on ARM config is disabled until an official build will be available. However the workflow is prepared to handle both Windows platform.

**Review by bryanpkc — APPROVED** [review]

**Comment by bryanpkc** [comment]
LGTM. I don't think the other maintainers have any objection to this PR.

**Pull request merged**
